app/assets/javascripts/oxalis/model/accessors/flycam_accessor.js

```diff
@@ -9,14 +9,61 @@ import type { Matrix4x4 } from "libs/mjs";
 import { M4x4 } from "libs/mjs";
 import * as THREE from "three";
 import { getMaxZoomStep } from "oxalis/model/accessors/dataset_accessor";
+import { getNecessaryBucketCount } from "oxalis/model/bucket_data_handling/data_rendering_logic";
+import memoizeOne from "memoize-one";
 
 // All methods in this file should use constants.PLANE_WIDTH instead of constants.VIEWPORT_WIDTH
 // as the area that is rendered is only of size PLANE_WIDTH.
 // If VIEWPORT_WIDTH, which is a little bigger, is used instead, we end up with a data texture
 // that is shrinked a little bit, which leads to the texture not being in sync with the THREEjs scene.
 
-// maximum difference between requested coordinate and actual texture position
-const MAX_ZOOM_STEP_DIFF = constants.MAX_RENDERING_TARGET_WIDTH / constants.PLANE_WIDTH;
+export function getUnzoomedBucketCountPerDim(
+  dataSetScale: Vector3,
+  forFallback: boolean = false,
+): Vector3 {
+  const baseVoxelFactors = scaleInfo.getBaseVoxelFactors(dataSetScale);
+  const necessaryVoxelsPerDim = baseVoxelFactors.map(f => f * constants.PLANE_WIDTH);
+
+  const factor = forFallback ? 2 : 1;
+  return necessaryVoxelsPerDim.map(v => Math.ceil(v / (factor * constants.BUCKET_WIDTH)));
+}
+
+function getUnzoomedBucketCountForRendering(dataSetScale: Vector3) {
+  const calculateBucketCountForOrtho = ([x, y, z]) => x * y + x * z + y * z;
+  const bucketsPerDim = getUnzoomedBucketCountPerDim(dataSetScale);
+  const fallbackBucketsPerDim = getUnzoomedBucketCountPerDim(dataSetScale, true);
+  return (
+    calculateBucketCountForOrtho(bucketsPerDim) +
+    calculateBucketCountForOrtho(fallbackBucketsPerDim)
+  );
+}
+
+// This function returns a value which indicates how much larger the rendered
+// plane can be than its original size **without** having to use the next
+// magnification. E.g., a value of two indicates that the viewport
+// can be 2 * viewport_width pixel wide while still being in zoom step 0.
+function unmemoizedCalculateMaxZoomStepDiff(dataSetScale: Vector3): number {
+  // TODO: Does sqrt make sense here?
+  const maxZoomStepDiff = Math.sqrt(
```

Comment:
The factor calculates how many more buckets can be passed to the GPU than necessary for rendering zoomstep = 1. Since the viewports are square, the amount of necessary buckets grows quadratic. So, taking the square root makes sense somehow, but I'm a bit fuzzy here after looking at this problem for several hours :)

Reply:
I think, sqrt makes sense since the function which defines the necessary amount of buckets is something like this: f(x) = (planeWidth / voxelPerBucket)**2 * (bucketsForX * bucketsForY + bucketsForX * bucketsForZ + bucketsForY * bucketsForZ).